src/adapter/src/coord/command_handler.rs

```diff
@@ -347,7 +347,23 @@ impl Coordinator {
         &mut self,
         user: &User,
         conn_id: &ConnectionId,
+        client_ip: &Option<IpAddr>,
     ) -> Result<RoleId, AdapterError> {
+        // Validate network policies for external users. Internal users
+        // can only connect on the internal interafces(internal http/
+        // pgwire). It is up to the person deploying the system to
+        // ensure these internal interfaces are well secured.
+        let system_config = self.catalog().state().system_config();
+        if !user.is_internal() {
+            let default_policy = NetworkPolicy::new(system_config.default_network_policy());
+            if let Some(ip) = client_ip {
```

Comment:
Remind me: what are the cases this could be `None` now? It's a little scary that if we accidentally stop plumbing the client IP here we'll stop enforcing network policies.

Reply:
(per live discussion: we think we can just return an `AdapterError::NetworkPolicyDenied("missing client IP")` if `client_ip` is `None` here, to fail closed instead of failing open)